iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/ImportInformationSpaceReducer.java

```diff
@@ -0,0 +1,368 @@
+        try {
+            super.cleanup(context);
+        } finally {
+            mos.close();
+        }
+    }
+    
+    @Override
+    public void reduce(ImmutableBytesWritable key, Iterable<InfoSpaceRecord> values, Context context)
+            throws IOException, InterruptedException {
+        byte[] idBytes = key.get();
+        Map<String, List<QualifiedOafJsonRecord>> mappedRecords = InfoSpaceRecordUtils.mapByColumnFamily(values);
+        
+        if (ByteArrayUtils.startsWith(idBytes, HBaseConstants.ROW_PREFIX_RESULT)) {
+            handleResult(idBytes, mappedRecords);
+        } else if (ByteArrayUtils.startsWith(idBytes, HBaseConstants.ROW_PREFIX_PERSON)) {
+            handleEntity(idBytes, mappedRecords.get(Type.person.name()), personConverter, outputNamePerson);
+        } else if (ByteArrayUtils.startsWith(idBytes, HBaseConstants.ROW_PREFIX_PROJECT)) {
+            handleEntity(idBytes, mappedRecords.get(Type.project.name()), projectConverter, outputNameProject,
+                    new RelationConversionDTO<ProjectToOrganization>(mappedRecords.get(projOrgColumnFamily),
+                            projectOrganizationConverter, outputNameProjectOrganization));
+        } else if (ByteArrayUtils.startsWith(idBytes, HBaseConstants.ROW_PREFIX_ORGANIZATION)) {
+            handleEntity(idBytes, mappedRecords.get(Type.organization.name()), organizationConverter, outputNameOrganization);
+        }
+    }
+    
+    // ------------------------ PRIVATE --------------------------
+    
+    /**
+     * Sets output directories.
+     * @param context hadoop context providing directories output names
+     */
+    private void setOutputDirs(Context context) {
+        Preconditions.checkNotNull(outputNameDocumentMeta = context.getConfiguration().get(OUTPUT_NAME_DOCUMENT_META),
+                "document metadata output name not provided!");
+        Preconditions.checkNotNull(outputNameDocumentProject = context.getConfiguration().get(OUTPUT_NAME_DOCUMENT_PROJECT),
+                "document project relation output name not provided!");
+        Preconditions.checkNotNull(outputNameProject = context.getConfiguration().get(OUTPUT_NAME_PROJECT),
+                "project output name not provided!");
+        Preconditions.checkNotNull(outputNamePerson = context.getConfiguration().get(OUTPUT_NAME_PERSON),
+                "person output name not provided!");
+        Preconditions.checkNotNull(outputNameDedupMapping = context.getConfiguration().get(OUTPUT_NAME_DEDUP_MAPPING),
+                "deduplication mapping output name not provided!");
+        Preconditions.checkNotNull(outputNameOrganization = context.getConfiguration().get(OUTPUT_NAME_ORGANIZATION),
+                "organization output name not provided!");
+        Preconditions.checkNotNull(outputNameProjectOrganization = context.getConfiguration().get(OUTPUT_NAME_PROJECT_ORGANIZATION),
+                "project to organization output name not provided!");
```

Comment:
`checkNotNull` returns the value and assignments in method arguments are confusing.
```
outputDocumentMeta = checkNotNull(context.getConfiguration().get(OUTPUT_NAME_DOCUMENT_META), "document metadata output name not provided!");
```